internal/client.go

```diff
@@ -835,6 +839,116 @@ type (
 		NonRetryableErrorTypes []string
 	}
 
+	// Priority contains metadata that controls relative ordering of task processing
+	// when tasks are backed up in a queue.
+	//
+	// Priority is attached to workflows and activities. By default, activities
+	// inherit Priority from the workflow that created them, but may override fields
+	// when an activity is started or modified.
+	//
+	// Despite being named "Priority", this message also contains fields that
+	// control "fairness" mechanisms.
+	//
+	// For all fields, the field not present or equal to zero/empty string means to
+	// inherit the value from the calling workflow, or if there is no calling
+	// workflow, then use the default value.
+	//
+	// For all fields other than fairness_key, the zero value isn't meaningful so
+	// there's no confusion between inherit/default and a meaningful value. For
+	// fairness_key, the empty string will be interpreted as "inherit". This means
+	// that if a workflow has a non-empty fairness key, you can't override the
+	// fairness key of its activity to the empty string.
+	//
+	// The overall semantics of Priority are:
+	//  1. First, consider "priority": higher priority (lower number) goes first.
+	//  2. Next, consider fairness: try to dispatch tasks for different fairness keys
+	//     in proportion to their weight.
+	//  3. Finally, tasks may be ordered by an additional ordering key.
+	//
+	// Applications may use any subset of mechanisms that are useful to them and
+	// leave the other fields to use default values.
+	//
+	// NOTE: Not all queues in the system may support the "full" semantics of all priority
+	// fields. (Currently only support in matching task queues is planned.)
```

Comment:
> (Currently only support in matching task queues is planned.)

Can we leave out server implementation details? The average SDK users does not know what a "matching task queue" is